inc/compat/plugins/compat-plugin-woo-save-abandoned-carts-pro.php

```diff
@@ -0,0 +1,99 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with plugin: CartBounty Pro - Save and recover abandoned carts for WooCommerce (by Streamline.lv).
+ */
+class FluidCheckout_WooSaveAbandonedCartsPro extends FluidCheckout {
+
+	/**
+	 * __construct function.
+	 */
+	public function __construct() {
+		$this->hooks();
+	}
+
+
+
+	/**
+	 * Initialize hooks.
+	 */
+	public function hooks() {
+		// Get plugin object instance
+		$class_object = $this->get_object_by_class_name_from_hooks( 'CartBounty_Pro_Public' );
+
+		// Bail if class object is not found in hooks
+		if ( ! $class_object ) { return; }
+
+		// Change the way input data is recovered from CartBounty to avoid conflicts with a similar feature from Fluid Checkout 
+		remove_filter( 'wp', array( $class_object, 'restore_input_data' ), 10 );
+		add_action( 'woocommerce_before_checkout_form', array( $this, 'maybe_restore_abandoned_cart_values_to_session' ), 10 );
```

Comment:
@vlad-dedikov see comment on Clickup task regarding use of `woocommerce_before_checkout_form `:
https://app.clickup.com/t/86dv38gwn?comment=90170079871420&threadedComment=90170080562243

Reply:
@diegoversiani I replaced the hook and found the reason for the checkout field data not being restored. More info regarding that is on Clickup.